LEGO1/mxeventpresenter.cpp

```diff
@@ -44,10 +47,71 @@ void MxEventPresenter::Destroy()
 
 	m_criticalSection.Enter();
 
-	if (m_unk50)
-		delete m_unk50;
+	if (m_data)
+		delete[] m_data;
 
 	Init();
 
 	m_criticalSection.Leave();
 }
+
+// OFFSET: LEGO1 0x100c2e30
+void MxEventPresenter::CopyData(MxStreamChunk* p_chunk)
+{
+	m_data = new MxU8[p_chunk->GetLength()];
+	memcpy(m_data, p_chunk->GetData(), p_chunk->GetLength());
+}
+
+// OFFSET: LEGO1 0x100c2e70
+void MxEventPresenter::ReadyTickle()
+{
+	MxStreamChunk* chunk = NextChunk();
+
+	if (chunk) {
+		CopyData(chunk);
+		m_subscriber->FUN_100b8390(chunk);
+		ParseExtra();
+		m_previousTickleStates |= 1 << (unsigned char) m_currentTickleState;
+		m_currentTickleState = TickleState_Starting;
+	}
+}
+
+// OFFSET: LEGO1 0x100c2eb0
+void MxEventPresenter::StartingTickle()
+{
+	MxStreamChunk* chunk = NextChunk();
+
+	if (chunk && m_action->GetElapsedTime() >= chunk->GetTime()) {
+		m_previousTickleStates |= 1 << (unsigned char) m_currentTickleState;
+		m_currentTickleState = TickleState_Streaming;
+	}
+}
+
+// OFFSET: LEGO1 0x100c2ef0
+undefined4 MxEventPresenter::PutData()
+{
+	MxAutoLocker lock(&m_criticalSection);
+
+	if (IsEnabled()) {
+		if (m_currentTickleState >= TickleState_Streaming &&
+			(m_currentTickleState <= TickleState_Repeating || m_currentTickleState == TickleState_Done)) {
+			if (m_currentChunk && m_currentChunk->GetLength()) {
+				if (m_data[12] == 2) {
```

Comment:
Note to self, not a problem: What value does this hardcoded 2 mean? Is this the start of a dictionary value?